File: MyNumberValidator.hs
import System.Environment (getArgs)
import Control.Monad (when)
import Text.Read (readMaybe)
import Data.Char (digitToInt)

checkMyNumber :: Int -> Bool
checkMyNumber mynumber
  | length (show mynumber) == 12 =
      let (checkDigit:p) = reverse $ map digitToInt $ show mynumber
          q     = [if n <= 6 then n + 1 else n - 5 | n <- [1..11]]
          x      = (`mod` 11) . sum . zipWith (*) q $ p
      in checkDigit == if x <= 1 then 0 else (11 - x)
  | otherwise = False

main :: IO()
main = do
  args <-getArgs
  when (not $ null args) $ do
    case readMaybe $ head args of
      Just myNumber -> let result = checkMyNumber myNumber
                       in
                         if result == True
                         then putStrLn "VALID"
                         else putStrLn "INVALID"
      Nothing -> putStrLn "Error: the argument must be a numeric number"

File: MyNumberValidator.py
#!/usr/bin/env python3

import sys

def checkMyNumber(mynumber):
    digits = [int(x) for x in list(str(mynumber))]
    digits.reverse()
    if (len(digits) != 12):
        return False
    checkDigit, *p = digits
    x = 0
    for i in range(1,12):
        q = i + 1 if (i <= 6) else i - 5
        x += p[i-1] * q
    x = x % 11
    if x <= 1:
        return checkDigit == 0
    else:
        return checkDigit == 11 - x

def main(argv):
    if len(argv) == 0:
        print("Error: empty arg")
        quit()
    try:
        mynumber = int(argv[0])
        print("VALID" if checkMyNumber(mynumber) else "INVALID")
    except ValueError:
        print("Error: the argument must be a numeric number")
        quit()

if __name__ == "__main__":
    main(sys.argv[1:])

